app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt

```diff
@@ -145,31 +148,43 @@ fun ProgressScreen(
             Spacer(modifier = Modifier.height(15.dp))
           }
 
+          item { DashBoardBarProgression() }
+
           item {
-            Text(
-                text = "Achievements",
-                fontSize = 18.sp,
-                fontWeight = FontWeight.Bold,
-                color = Color.Black,
-                textAlign = TextAlign.Start,
-                modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp).fillMaxWidth())
-          }
+            when (uiState.collectAsState().value) {
+              DashboardStateProgression.Achievement -> {
 
-          if (currentProgression.achievements.isEmpty()) {
-            item {
-              Text(
-                  text = "You don't have any achievements yet!",
-                  fontSize = 15.sp,
-                  fontWeight = FontWeight.Normal,
-                  color = ColorPalette.SECONDARY_TEXT_COLOR,
-                  modifier = Modifier.padding(top = 10.dp).testTag("emptyAchievementsText"))
-            }
-          } else {
-            itemsIndexed(currentProgression.achievements) { index, achievementName ->
-              val achievementEnum = enumValueOf<MedalsAchievement>(achievementName)
-              Box(modifier = Modifier.testTag("achievementItem${index}")) {
-                AchievementItem(achievementEnum.achievement)
-                HorizontalDivider(thickness = 0.5.dp, color = Color.LightGray)
+                if (currentProgression.achievements.isEmpty()) {
+
+                  Text(
+                      text = "You don't have any achievements yet!",
+                      fontSize = 15.sp,
+                      fontWeight = FontWeight.Normal,
+                      color = ColorPalette.SECONDARY_TEXT_COLOR,
+                      modifier = Modifier.padding(top = 10.dp).testTag("emptyAchievementsText"))
+                } else {
+
+                  Column {
+                    currentProgression.achievements.forEach { achievementName ->
+                      val achievementEnum = enumValueOf<MedalsAchievement>(achievementName)
+                      Box(modifier = Modifier.testTag("achievementItem")) {
+                        AchievementItem(achievementEnum.achievement, false)
+                        HorizontalDivider(thickness = 0.5.dp, color = Color.LightGray)
+                      }
+                    }
+                  }
+                }
+              }
+              DashboardStateProgression.Training -> {
+
+                enumValues<ExerciseAchievement>().forEach {
+                  it.achievement.description =
+                      "Record : 0.0 sec" // TODO: use the progression MVVM with current record
+                  Box(modifier = Modifier.testTag("exerciseItem" + it.name)) {
```

Comment:
I personally find `"exerciseItem${it.name}"` cleaner, but that's just a suggestion, fell free to leave it like that.

Reply:
Done, thank you!